soc/st/stm32/stm32mp13x/soc.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for STM32MP13 processor
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/linker/linker-defs.h>
+
+#include <stm32_ll_bus.h>
+#include <cmsis_core.h>
+
+/**
+ * @brief Perform basic hardware initialization at boot.
+ *
+ * This needs to be run from the very beginning.
+ * So the init priority has to be 0 (zero).
+ *
+ * @return 0
+ */
+#define VECTOR_ADDRESS ((uintptr_t)_vector_start)
+
+void relocate_vector_table(void)
+{
+	write_sctlr(read_sctlr() & ~HIVECS);
+	write_vbar(VECTOR_ADDRESS & VBAR_MASK);
+	barrier_isync_fence_full();
+}
+
+static int stm32mp13_a7_init(void)
+{
+	/* Update CMSIS SystemCoreClock variable (HCLK) */
+	SystemCoreClock = 1000000000U;
+
+	__ASM volatile(
+
+		/* Read CP15 System Control register */
+		"MRC     p15, 0, R0, c1, c0, 0\n"
+
+		/* Clear TE bit to take exceptions in Thumb mode to fix the DDR init */
+		"BIC     R0, R0, #(0x1 << 30)\n"
+
+		/* Write value back to CP15 System Control register */
+		"MCR     p15, 0, R0, c1, c0, 0\n"
+		"ISB\n"
+		"LDR     R0, =_vector_table\n"
+		"MCR     p15, 0, R0, c12, c0, 0\n");
+	return 0;
+}
+
+static const struct arm_mmu_region mmu_regions[] = {
+	MMU_REGION_FLAT_ENTRY("vectors", 0xC0000000, 0x1000, MPERM_R | MPERM_X | MT_NORMAL),
+
+	MMU_REGION_FLAT_ENTRY("GIC", 0xA0021000, 0x7000, MPERM_R | MPERM_W | MT_DEVICE),
+
+	MMU_REGION_FLAT_ENTRY("APB1", 0x40010000, 0x19400, MPERM_R | MPERM_W | MT_DEVICE),
+
+	MMU_REGION_FLAT_ENTRY("APB2", 0x44000000, 0x14000, MPERM_R | MPERM_W | MT_DEVICE),
+
+	MMU_REGION_FLAT_ENTRY("AHB2", 0x48000000, 0x400, MPERM_R | MPERM_W | MT_DEVICE),
+
+	MMU_REGION_FLAT_ENTRY("APB6", 0x4C000000, 0xC400, MPERM_R | MPERM_W | MT_DEVICE),
+
+	MMU_REGION_FLAT_ENTRY("AHB4", 0x50000000, 0xD400, MPERM_R | MPERM_W | MT_DEVICE),
```

Comment:
s/0xD400/0x1D400/

Reply:
@etienne-lms 
seems to me that is `0xD400`. last peripheral on this bus is EXTI(0x5000D000 - 0x5000D3FF)

Reply:
Sorry, you're right. `0xD400`